### ISSUE TITLE
--gate-endpoint flag not working with URIs anymore

### ISSUE THREAD

**Issue by MrTrustor** [spinnaker/spin#163]
#152 has changed the behaviour of the `--gate-endpoint` flag.

I have a [setup](https://github.com/spinnaker/spinnaker/blob/master/solutions/kayenta/ci/scripts/first-deployment.sh#L22) where gate is available on `http://localhsot:8080/gate`. This used to work fine until Spin 1.7.0. It's now failing with the following error:

```
$ spin application save --application-name sampleapp --owner-email example@example.com --cloud-providers kubernetes --gate-endpoint http://localhost:8080/gate
Could not dial http client, please ensure Gate is running. Failing.
dial tcp: address tcp/8080/gate: unknown port
```

**Comment by jtk54** [comment]
We've seen some issues with that change as well. I'll rework or revert it.

**Comment by jtk54** [comment]
Merged and cherry picked into 1.13 branch, we'll release a patch within a week or so.
